formats/notion/convert-to-md.ts

```diff
@@ -19,226 +23,226 @@ export function convertNotesToMd({
+		// });
 
-		fileInfo.body = convertLinksToObsidian(fileInfo.body, {
+		convertLinksToObsidian(fileInfo.notionLinks, {
 			idsToFileInfo,
 			pathsToAttachmentInfo,
 			attachmentFolderPath,
 			fileInfo,
 		});
 
-		replaceNestedFormatting(fileInfo.body);
-		replaceNewlinesInFormatting(fileInfo.body);
-		encodeNewlines(fileInfo.body);
-		stripLinkImages(fileInfo.body);
+		replaceNestedTags(fileInfo.body, 'strong');
+		replaceNestedTags(fileInfo.body, 'em');
+		stripLinkFormatting(fileInfo.body);
+		encodeNewlinesToBr(fileInfo.body);
+		splitBrsInFormatting(fileInfo.body, 'strong');
+		splitBrsInFormatting(fileInfo.body, 'em');
 		fixNotionDates(fileInfo.body);
 		fixNotionLists(fileInfo.body);
 		replaceTableOfContents(fileInfo.body);
-		fileInfo.body = htmlToMarkdown(fileInfo.body);
-		fileInfo.body = escapeHashtags(fileInfo.body);
+
+		fileInfo.markdownBody = htmlToMarkdown(fileInfo.body.innerHTML);
+		fileInfo.markdownBody = escapeHashtags(fileInfo.markdownBody);
 
 		if (fileInfo.properties) {
-			fileInfo.yamlProperties = fileInfo.properties
-				.filter((property) => property.content)
-				.map((property) =>
-					convertPropertyToYAML(property, {
-						fileInfo,
-						idsToFileInfo,
-						pathsToAttachmentInfo,
-						attachmentFolderPath,
-					})
-				);
+			fileInfo.yamlProperties = fileInfo.properties.map((property) =>
+				convertPropertyToYAML(property, {
+					fileInfo,
+					idsToFileInfo,
+					pathsToAttachmentInfo,
+					attachmentFolderPath,
+				})
+			);
 		}
 	}
 }
 
-const replaceNestedFormatting = (body: string) => {
-	body = body
-		.replace(/<strong>(<strong>)+/g, '<strong>')
-		.replace(/<\/strong>(<\/strong>)+/g, '</strong>')
-		.replace(/<em>(<em>)+/g, '<em>')
-		.replace(/<\/em>(<\/em>)+/g, '</em>');
+const replaceNestedTags = (body: HTMLElement, tag: 'strong' | 'em') => {
+	body.querySelectorAll(tag).forEach((el) => {
+		const nestedEls = el.querySelectorAll(tag);
+		if (nestedEls.length === 0) return;
+		const textContent = el.textContent;
+		nestedEls.forEach((nestedEl) => nestedEl.remove());
+		el.setText(textContent);
+	});
 };
 
-const replaceNewlinesInFormatting = (body: string) => {
-	const strongs = body.matchAll(/<strong>((.|\n)*)<\/strong>/g);
-	for (let strong of strongs) {
-		if (strong[1].contains('\n')) {
-			const strongs = strong[1].split('\n');
-			body = body.replace(
-				strong[0],
-				strongs.map((strong) => `<strong>${strong}</strong>`).join('\n')
-			);
-		}
-	}
-	const italics = body.matchAll(/<em>((.|\n)*)<\/em>/g);
-	for (let italic of italics) {
-		if (italic[1].contains('\n')) {
-			const italics = italic[1].split('\n');
-			body = body.replace(
-				italic[0],
-				italics.map((italic) => `<em>${italic}</em>`).join('\n')
-			);
-		}
-	}
+const splitBrsInFormatting = (body: HTMLDivElement, tag: 'strong' | 'em') => {
+	body.querySelectorAll(tag).forEach((el) => {
+		// TODO: check to see if newlines are stripped in textContent
+		if (!/<br \/>/g.test(el.textContent)) return;
```

Comment:
Any chance we can do this by manipulating/splicing DOM elements instead of relying on textContent and innerHTML?

Reply:
Also this doesn't work with more than one level of nesting right?

Reply:
Honestly that seems way too complicated for me to do concisely. It worked perfectly with Regex, so I'm just going to do this transformation after all the DOM mutations on the HTML text.